### PR TITLE
Fix MatMul SmartReshape in case of 1D input

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
Handled a case when "Other" input to MatMul is 1D

### Tickets:
 - *CVS-141638*

### CODE REVIEW

File: src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp [hunk 1]
```diff
@@ -36,7 +36,16 @@ bool relax_hc_reshape_followed_by_matmul(const ov::pass::pattern::PatternValueMa
         // avoiding loop creation
         return false;
 
-    const auto idx = reshape_is_A_input ? (matmul->get_transpose_b() ? -1 : -2) : (matmul->get_transpose_a() ? -2 : -1);
+    if (shape_source.get_partial_shape().rank().is_dynamic()) {
+        return false;
+    }
```
Comment (jane-intel) [inline src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp:41]: let's put it in the pattern.

**Review by jane-intel — APPROVED** [review]

File: src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp (hunk 1 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp]: done

File: src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp [hunk 2]
```diff
@@ -53,15 +58,17 @@ bool relax_hc_reshape_followed_by_matmul(const ov::pass::pattern::PatternValueMa
...
 
 }  // namespace
 
 ov::pass::ReshapeAMatMul::ReshapeAMatMul() {
     MATCHER_SCOPE(ReshapeAMatMul);
-    auto other_input_label = pattern::any_input();
+    auto other_input_predicate = [](ov::Output<ov::Node> output) -> bool {
+        return output.get_partial_shape().rank().is_static();
+    };
+    auto other_input_label = pattern::any_input(other_input_predicate);
```
Comment (jane-intel) [inline src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp:71]:
```suggestion
    auto other_input_label = pattern::any_input(pattern::has_static_rank());
```

**Review by jane-intel — CHANGES_REQUESTED** [review]

File: src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp [hunk 3]
```diff
@@ -86,7 +93,10 @@ ov::pass::ReshapeAMatMul::ReshapeAMatMul() {
 
 ov::pass::ReshapeBMatMul::ReshapeBMatMul() {
     MATCHER_SCOPE(ReshapeBMatMul);
-    auto other_input_label = pattern::any_input();
+    auto other_input_predicate = [](ov::Output<ov::Node> output) -> bool {
+        return output.get_partial_shape().rank().is_static();
+    };
+    auto other_input_label = pattern::any_input(other_input_predicate);
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp:99]: I think we should make use of what @jane-intel has suggested before.

File: src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp (hunk 2 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp]: done

File: src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp (hunk 3 above)
Comment (itikhono) [reply src/common/transformations/src/transformations/smart_reshape/matmul_sr.cpp]: done

**Review by jane-intel — APPROVED** [review]

**Review by CuriousPanCake — APPROVED** [review]

**Pull request merged**